src/provider/meta.test.ts

```diff
@@ -0,0 +1,70 @@
+import { ExternalProvider, JsonRpcProvider } from '@ethersproject/providers'
+import WalletConnectProvider from '@walletconnect/ethereum-provider'
```

Comment:
My overall comment is whether this should live in `conedison` instead of `web3-react`. Seems to be adding an unnecessary coupling between `conedison` and `walletconnect` that can get out of control unless we maintain both and keep them in sync. 

In theory, this will require an update to WalletConnect 2.x once we ship a new `web3-react` version and make working locally a bit harder, as this package won't be compatible.

I was thinking about adding `utils` or `core` package to `web3-react` for a long time now, with this (and a couple of other helpers) being good candidates for first round of utilities.

Maybe it's not too late to reconsider and submit this package against `web3-react`, or merge this "as is", and provide a new `v2` compatible version of this file inside `web3-react` going forward.

Reply:
In practice, I now need to upgrade this & web3-react at the same time and symlink two packages in order to develop wc2 in interface.

Reply:
Yes, you'll need to upgrade these at the same time, because the whole dApp will need to upgrade to v2 at the same time. I don't think this is an issue, as long as they support the same methods (I'll check that shortly).

I don't think this should be a part of web3-react. It is a separate utility that we use on our own interface. Users can continue to use WC interfaces directly through the API exposed through web3-react, and I don't think we should complicate the abstraction by adding utility libraries on top of it.

Reply:
Makes sense, thanks for clarification!